src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -389,14 +389,29 @@ internal bool RunLocally(string code, out string errorMessage, bool suppressOutp
         try
         {
             Pwsh.Commands.AddCommand(command);
-            Pwsh.AddCommand(_outDefaultCommand);
 
             if (!suppressOutput)
             {
                 Pwsh.Commands.Commands[0].MergeMyResults(PipelineResultTypes.Error, PipelineResultTypes.Output);
             }
 
-            Pwsh.InvokeAndClear();
+            var result = Pwsh.InvokeAndClearWithResult();
+
+            foreach (var item in result)
+            {
+                var value = item?.BaseObject ?? item;
+
+                if (value is string str)
+                {
+                    var formatted = new FormattedValue("text/plain", str + Environment.NewLine);
+                    KernelInvocationContext.Current?.Publish(
```

Comment:
It's best to avoid calling `KernelInvocationContext.Current` from within .NET Interactive code. This method is really meant as a convenience for notebook users but if possible it's better to use a `KernelInvocationContext` object directly.

One is passed into `HandleAsync` for the main code path, so passing it here as an optional parameter will solve that.

`SecretManager` also calls this method but it's not expecting to display anything directly, so I would say if context isn't passed, don't bother trying to display anything.

Reply:
updated, added a context as a parameter and using it now